### PR TITLE
Implement Component Manager

### PR DESCRIPTION
**Pull request by AlexJump24**
**Issue:** https://github.com/cachethq/core/issues/71

I have the functionality of this working, but leaving in WIP for now as I write some tests. I am quite unfamiliar with working with Testbench so not sure if I have set this up correctly after some research and delving into which service providers I needed loading, my initial tests are passing though.

The Filament integration was surprisingly tricky due to the nesting, I'm not sure if there is a better way to approach this, but I have built up the state by mapping the array of models to an associative array which Filament was able to nicely hook into.  

image

I also have a bit of an annoying issue when both sections are collapsed, which I don't think looks too great and I'll need to look into.

image

### CODE REVIEW

File: src/Filament/Widgets/Components.php [hunk 1]
```diff
@@ -2,11 +2,78 @@
...
+                                            ->columns(count(ComponentStatusEnum::cases()))
+                                            ->live()
+                                            ->options(ComponentStatusEnum::class)
+                                            ->afterStateUpdated(function (ComponentStatusEnum $state) use ($component) {
+                                                return $component->update(['status' => $state]);
+                                            })
+                                    ]);
+                            })->toArray();
+                    })
+                    ->collapsed($componentGroup->isCollapsible());
```
Comment (joelbutcher) [inline src/Filament/Widgets/Components.php:68]: Just a hunch, but what happens if you add `->columnSpanFull()` at the end of this? Does this fix your collapsed state issue?

```suggestion
                    ->collapsed($componentGroup->isCollapsible())
                    ->columnSpanFull();
```

**Review by joelbutcher — COMMENTED** [review]
Not entirely sure if my comment is useful. 

Also noticed that this form has no actions, it might be worth instead investigating using infolists instead of form components.

**Comment by jbrooksuk** [comment]
@AlexJump24 this is awesome, thank you!

File: src/Filament/Widgets/Components.php (hunk 1 above)
Comment (AlexJump24) [reply src/Filament/Widgets/Components.php:68]: Yeah good call, I had tried that but it didn't work, then woke up this morning and realised I could just set the form to be full width and it has fixed it 😄

**Comment by AlexJump24** [comment]
Just realised I need to pull in components not attached to component groups, I will try and look at that soon as well.

**Comment by jbrooksuk** [comment]
@AlexJump24 I was just about to comment the same thing :) Also, I've changed from using `Radio` to `ToggleButtons` as I feel this is more in keeping with the rest of the UI. Not totally sold on it yet though, so happy to revert that back.

**Comment by AlexJump24** [comment]
> @AlexJump24 I was just about to comment the same thing :) Also, I've changed from using `Radio` to `ToggleButtons` as I feel this is more in keeping with the rest of the UI. Not totally sold on it yet though, so happy to revert that back.

Nice! Yeah that seems much better to me! Especially it being a dashboard its good to stand out a bit more I think as well.

**Comment by jbrooksuk** [comment]
I didn't update the tests 😅

**Comment by AlexJump24** [comment]
I've made an update, I'm not sure of it tbh. I had to create a default model. I also had to take the component groups off of the public Livewire property because of how Livewire rehydrates was causing all sorts of issues.

image

See default now shows even here (I'm not sure if thats ok?) 
image

**Comment by jbrooksuk** [comment]
This is fine with me, but I think for it to be a worthwhile change, the status page controller needs to respect the "default" group.

**Comment by jbrooksuk** [comment]
@AlexJump24 I'm playing around with ungrouping ungrouped components in the status page:

CleanShot 2024-10-09 at 20 33 48@2x

You know, I think I quite like this approach. If you want them grouped, then create a group - right? So, I guess the same applies to the dashboard - we just show each ungrouped component individually.

**Comment by jbrooksuk** [comment]
@AlexJump24 I think we can tweak the component a little more to make it cleaner and customizable.

image

The changes here:

1. `components.blade` widget can just be the `x-filament-widgets::widget` and the form. No need for the section inside.
2. Make sections collapsible and persist their collapsed state.
3.

**Comment by AlexJump24** [comment]
Done, cheers yeah thats much better.

**Comment by jbrooksuk** [comment]
LGTM! What's left on this before we get it reviewed and merged? :)

**Comment by AlexJump24** [comment]
@jbrooksuk was just wondering if we should lose the label on the ungrouped components as its duplicated from the section title. Otherwise I think its ready to be merged.

**Comment by jbrooksuk** [comment]
Yeah, let's do that to remove the redundant duplication and save some space.

**Review by jbrooksuk — APPROVED** [review]

**Pull request merged**